### ISSUE TITLE
Sync not maintaining file structure

### ISSUE THREAD

**Issue by hmorzaria** [Azure/azure-storage-azcopy#1666]
### Which version of the AzCopy was used? 
10.13
### Which platform are you using? (ex: Windows, Mac, Linux)
Microsoft Azure virtual machine
Linux, Ubuntu 18.04 Server

### What command did you run?
`sudo azcopy sync "https://<mystorageaccount>/mycontainerdata?<mySAS>" "/home/myrootdir" --mirror-mode=true --recursive`

`sudo azcopy sync "https://<mystorageaccount>/mycontainerdata?<mySAS/>" "/home/myrootdir/mycontainerdata" --mirror-mode=true --recursive`

### What problem was encountered?
I had already previously copied the data in a folder also named mycontainerdata.
When running 
`sudo azcopy sync "https://<mystorageaccount>/mycontainerdata?<mySAS>" "/home/myrootdir" --mirror-mode=true --recursive`
a folder named mycontainer_data is automatically created and all the data is downloaded again. 

If I run
`sudo azcopy sync "https://<mystorageaccount>/mycontainerdata?<mySAS/>" "/home/myrootdir/mycontainerdata" --mirror-mode=true --recursive`
A new subfolder named mycontainer_data is automatically created and all the data is downloaded again. 

If I change the name of the destination directory to mynewcontainerdata, and specify that directory a new subfolder named mycontainer_data is automatically created and all the data is downloaded again. 

How to prevent all the contents of the container from being downloaded every time? should I have used sync the first time rather than copy? (which is not intuitive). Seems for this feature to be useful, contents of whatever local folder is specified should be synced with whatever container is specified.

**Comment by hmorzaria** [comment]
I get the same behavior when not using a SAS token, directly logging into Azure Storage AzCopy. If I run two consecutive sync operations on the same container, the second one does act as expected and doesn't transfer any files. If I run a sync or copy operation in between for a separate container it then seems to "forget" which folder is synced

**Comment by hmorzaria** [comment]
This seems to be the same behavior flagged by the issue AzCopy sync re-copies files even if file already exists and is unchanged (date modified matches). #1631

**Comment by hmorzaria** [comment]
tested in a Windows machine running Az copy 10.10.0 and I don't see the same behavior.  
`sudo azcopy sync "https://<mystorageaccount>/mycontainerdata?<mySAS/>" "/home/myrootdir/mycontainerdata" --recursive` with a previously exisiting directory will accurately compare folder contents and not download the whole folder again.

**Comment by adreed-msft** [comment]
Does the issue persist without --mirror-mode? This sounds like a bug.

**Comment by hmorzaria** [comment]
@adreed-msft Yes, the issue persists even without mirror mode, I have also updated to Azcopy 10.14.1 and I get the same behavior

**Comment by g7cnp** [comment]
I've been trying to distribute a folder of documents to several workstations, so the source is an Azure file share, and the destination is a folder on the workstation running the azcopy command. I may occasionally update or add documents to the source folder, so I rerun the command once a day to pick up any new docs or any with a later modified date.
I'm not bothered about removing files so I have a choice.
I can use the **sync** command or, as mentioned in the docs **copy --overwrite=ifSourceNewer** 

Whilst looking at the destination folder this appears to work fine, new files are added, and updated files are updated, and even files updated at the destination are preserved. The issue is files that are unchanged at either end seem to get copied anyway even though there is no need, this is wasteful and time consuming and stops the solution scaling to large numbers of files.

It is as if when comparing the file dates to determine which to copy it's doing,
if (source.date -ge destination.date) {copy file} 
rather than
if (source.date -gt destination.date) {copy file} 
Please excuse the Pseudo code :-) 
I had given up on this but will double check with 10.14.1

**Comment by g7cnp** [comment]
Yes i'm getting the same behaviour with 10.14.1

using sync
**First Run (Empty destination)**
Job 91bd5a8e-5950-3848-6de4-5c4ba3d931e1 Summary
Files Scanned at Source: 6
Files Scanned at Destination: 0
Elapsed Time (Minutes): 0.0334
Number of Copy Transfers for Files: 6
Number of Copy Transfers for Folder Properties: 0
Total Number Of Copy Transfers: 6
Number of Copy Transfers Completed: 6
Number of Copy Transfers Failed: 0
Number of Deletions at Destination: 0
Total Number of Bytes Transferred: 39877
Total Number of Bytes Enumerated: 39877
Final Job Status: Completed

**Second Run (no files changed)**
Job b266d983-ef2f-6c44-4a0b-34175ad543e7 Summary
Files Scanned at Source: 6
Files Scanned at Destination: 6
Elapsed Time (Minutes): 0.0335
Number of Copy Transfers for Files: 6
Number of Copy Transfers for Folder Properties: 0
Total Number Of Copy Transfers: 6
Number of Copy Transfers Completed: 6
Number of Copy Transfers Failed: 0
Number of Deletions at Destination: 0
Total Number of Bytes Transferred: 39877
Total Number of Bytes Enumerated: 39877
Final Job Status: Completed

I would expect to see 
Number of Transfers Skipped: 6
TotalBytesTransferred: 0
As nothing needs updating?

**Comment by bricelabelle** [comment]
I opened a ticket with Microsoft, went through multiple stages of reproducing this issue for them, and they finally acknowledged it as a bug that they were able to reproduce. They said a fix would make it into a future release of azcopy, but that was a few months ago so I came on here to see if any progress had been made.

**Comment by bitzertech** [comment]
This is still an issue with azcopy 10.20.1
